presto-main/src/main/java/com/facebook/presto/operator/aggregation/StatisticalDigestAggregationFunction.java

```diff
@@ -0,0 +1,271 @@
+import static com.facebook.presto.spi.type.StandardTypes.TDIGEST;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
+import static java.lang.invoke.MethodHandles.insertArguments;
+
+public abstract class StatisticalDigestAggregationFunction
+        extends SqlAggregationFunction
+{
+    private static final long DEFAULT_WEIGHT = 1L;
+    private final String name;
+    private final String type;
+    private final StatisticalDigestStateFactory factory;
+
+    private static final MethodHandle T_DIGEST_INPUT_DOUBLE = methodHandle(TDigestAggregationFunction.class,
+            "inputDouble",
+            StatisticalDigestState.class,
+            double.class,
+            long.class,
+            double.class);
+    private static final MethodHandle QUANTILE_DIGEST_INPUT_DOUBLE = methodHandle(
+            QuantileDigestAggregationFunction.class,
+            "inputDouble",
+            StatisticalDigestState.class,
+            double.class,
+            long.class,
+            double.class);
+    private static final MethodHandle INPUT_REAL = methodHandle(QuantileDigestAggregationFunction.class,
+            "inputReal",
+            StatisticalDigestState.class,
+            long.class,
+            long.class,
+            double.class);
+    private static final MethodHandle INPUT_BIGINT = methodHandle(
+            QuantileDigestAggregationFunction.class,
+            "inputBigint",
+            StatisticalDigestState.class,
+            long.class,
+            long.class,
+            double.class);
+
+    private static final MethodHandle COMBINE_FUNCTION = methodHandle(
+            StatisticalDigestAggregationFunction.class,
+            "combineState",
+            StatisticalDigestState.class,
+            StatisticalDigestState.class);
+    private static final MethodHandle OUTPUT_FUNCTION = methodHandle(
+            StatisticalDigestAggregationFunction.class,
+            "evaluateFinal",
+            StatisticalDigestStateSerializer.class,
+            StatisticalDigestState.class,
+            BlockBuilder.class);
+
+    StatisticalDigestAggregationFunction(String name, String type, StatisticalDigestStateFactory factory, boolean hidden, TypeSignature... typeSignatures)
+    {
+        super(new Signature(
+                        name,
+                        AGGREGATE,
+                        ImmutableList.of(comparableTypeParameter("V")),
+                        ImmutableList.of(),
+                        parseTypeSignature(type + "(V)"),
+                        ImmutableList.copyOf(typeSignatures),
+                        false),
+                hidden);
+        this.name = name;
+        this.type = type;
+        this.factory = factory;
+    }
+
+    @Override
+    public String getDescription()
```

Comment:
Remove this. All children class provides description anyways and you probably want to enforce that.